src/common/userlist.c

```diff
@@ -130,8 +130,12 @@ userlist_add_hostname (struct session *sess, char *nick, char *hostname,
 			g_free (user->hostname);
 			user->hostname = g_strdup (hostname);
 		}
-		if (!user->realname && realname && *realname)
+		if (realname && (!user->realname || strcmp(user->realname, realname)))
```

Comment:
```suggestion
        if (realname && g_strcmp0 (user->realname, realname) != 0)
```

Reply:
I'm not sure if the empty string check is necessarily what you expect, but it keeps previous behavior.